programs/mango-v4/src/state/bank.rs

```diff
@@ -955,6 +966,23 @@ impl Bank {
         Ok(state.price)
     }
 
+    /// Tries to return the primary oracle price, and if there is a confidence or staleness issue returns the fallback oracle price.
+    pub fn oracle_price_with_fallback(
+        &self,
+        oracle_acc: &impl KeyedAccountReader,
+        fallback_oracle_acc_opt: Option<&impl KeyedAccountReader>,
+        staleness_slot: Option<u64>,
+    ) -> Result<I80F48> {
+        let primary_price = self.oracle_price(oracle_acc, staleness_slot);
+        if primary_price.is_ok() || fallback_oracle_acc_opt.is_none() {
```

Comment:
Users could pass the wrong account for `oracle_acc` here, then the fallback would be used independently of the state of the main oracle!

There's different ways to go about this: I think it's appropriate here to check that the actual error returned is one of the ones that should enable going to a fallback oracle.

Note also that check_confidence_and_maybe_staleness() does some logging on failure that might be inappropriate or at least too expensive (didn't check) to do for all failures.